public/app/plugins/datasource/prometheus/types.ts

```diff
@@ -18,6 +18,7 @@ export interface PromQuery extends DataQuery {
   requestId?: string;
   showingGraph?: boolean;
   showingTable?: boolean;
+  queryPreview?: boolean;
```

Comment:
Just an idea - wouldn't it make sense to group editor props into an object such as: 
```
editorSettings?: {
  queryPreview?: boolean;
  editorMode?: QueryEditorMode;
}
```
So it is clear which ones are used to set the ui of editor and which are actual query?

Reply:
yes!

Reply:
I tried this but nesting options makes the model just more complex, the typings more complex (cue schema),  changing a editor setting becomes more complex (more nested spread operators). 

So opted instead to prefix editor settings with "editor" , so we have "editorPreview" and "editorMode" 
```ts
export interface PromQuery extends DataQuery {
  ....
  showingTable?: boolean;
  /** Code, Builder or Explain */
  editorMode?: QueryEditorMode;
  /** Controls if the query preview is shown */
  editorPreview?: boolean;
  ....
}
```

Reply:
👍 good idea!